### PR TITLE
add test for null type

### PR DESCRIPTION
**Pull request by SeriyBg**

### CODE REVIEW

**Review by Fly-Style — APPROVED** [review]

File: src/test/java/com/hazelcast/jdbc/DriverTypeConversionTest.java [hunk 1]
```diff
@@ -680,6 +680,21 @@ void shouldSearchForComparableObjects() throws SQLException {
         assertThat(resultSet.next()).isTrue();
     }
 
+    @Test
+    void shouldSearchForNullValues() throws SQLException {
+        Person person = new Person("Jack", 20);
+        IMap<Object, Object> types = member.getMap("person");
+        types.put(1, person);
+
+        PreparedStatement preparedStatement = connection.prepareStatement("SELECT CAST(? AS OBJECT), name, age FROM person");
```
Comment (viliam-durina) [inline src/test/java/com/hazelcast/jdbc/DriverTypeConversionTest.java:689]: Why do we need the `CAST`?

Comment (SeriyBg) [reply src/test/java/com/hazelcast/jdbc/DriverTypeConversionTest.java:689]: Simply `SELECT ?` is not allowed and throws an exception `Illegal use of dynamic parameter`

**Review by viliam-durina — APPROVED** [review]
